src/widgets/medications/medication-level-two.component.tsx

```diff
@@ -50,43 +51,142 @@ export default function MedicationLevelTwo(props: MedicationsOverviewProps) {
+                            {dayjs(medication.dateActivated).format(
+                              "DD-MMM-YYYY"
+                            )}
+                          </td>
+                        </tr>
+                      </React.Fragment>
+                    );
+                  })}
+            </tbody>
           </table>
+          <div className={styles.medicationFooter}>
+            <p
+              style={{ color: "var(--omrs-color-ink-medium-contrast)" }}
+              className={"omrs-type-body-large"}
+            >
+              No more medications available.
+            </p>
+          </div>
         </SummaryCard>
       </React.Fragment>
     );
   }
 
-  function displayPreviousMedication() {
-    <SummaryCard
-      name={t("Medications-previous", "Medications-previous")}
-      match={props.match}
-      styles={{ width: "90%" }}
-      addBtnUrl={`/patient/${patientUuid}/chart/add`}
-    >
-      <table className={styles.medicationsTable}>
-        <thead>
-          <tr>
-            <td>Name</td>
-            <td>
-              <div className={styles.centerItems}>Status</div>
-            </td>
-            <td>Start Date</td>
-          </tr>
-        </thead>
-        <tbody>{patientMedications && parseMedication()}</tbody>
-      </table>
-      <div className={styles.medicationFooter}>
-        <p
-          style={{ color: "var(--omrs-color-ink-medium-contrast)" }}
-          className={"omrs-type-body-large"}
+  function displayPastMedications() {
+    return (
+      <React.Fragment>
+        <SummaryCard
+          name={t("Medications - past", "Medications - past")}
+          match={props.match}
+          styles={{ width: "90%" }}
+          addBtnUrl={`/patient/${patientUuid}/chart/add`}
         >
-          No more medication available
-        </p>
-      </div>
-    </SummaryCard>;
+          <table className={styles.medicationsTable}>
+            <thead>
+              <tr>
+                <td>Name</td>
+                <td>
+                  <div className={styles.centerItems}>Status</div>
+                </td>
+                <td>Start Date</td>
+              </tr>
+            </thead>
+            <tbody>
+              {patientMedications &&
+                patientMedications
+                  .filter(med => med.action !== "NEW")
+                  .map(medication => {
+                    return (
+                      <React.Fragment key={medication.uuid}>
+                        <tr>
+                          <td>
+                            <span
+                              style={{
+                                fontWeight: 500,
+                                color: "var(--omrs-color-ink-high-contrast)"
+                              }}
+                            >
+                              {medication.drug.display}
+                            </span>
+                            {" \u2014 "} {medication.doseUnits.display}{" "}
```

Comment:
why so many whitespaces? is there a different way we can achieve this without having the whitespace inside html?

Reply:
Removed most of these